File: AngularAndNetCoreAuth/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
namespace AngularAndNetCoreAuth.Models
{
    public class Product:BaseEntity
    {
      
        // refers to ID in srms student table
        [Required]
        public string Code { set; get; }
        [Required]
        public string Name { set; get; }
        [Required]
        public bool AcceptPartPayment { set; get; }
        [Required]
        public string Purpose { set; get; }
        [Required]
        public decimal Amount { set; get; }
        [Required]
        public string Currency { set; get; }
        [Required]
        [DataType(DataType.Date)]
        public DateTime StartDate { get; set; }
        [Required]
        [DataType(DataType.Date)]
        public DateTime EndDate { get; set; }
        [Required]
        public string Banks { set; get; }
        [Required]
        public string Instructions { set; get; }
        [Required]
        public bool Status { set; get; }
      
        public string Url { set; get; }
    }
}

File: New_Service/backend/src/main/java/com/ttu/pay/model/Product.java
package com.ttu.pay.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;
import java.util.UUID;
@Getter
@Setter
@Accessors(chain=true)
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Product implements Serializable {
    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private UUID id;
    @NotNull
    @JsonProperty("code")
    private String code;
    @NotNull
    @JsonProperty("name")
    private String name;
    @NotNull
    @JsonProperty("part")
    private boolean acceptPartPayment=true;
    @NotNull
    private String purpose;
    @NotNull
    private BigDecimal amount;
    @NotNull
    private String currency;
    @NotNull
    private LocalDate startDate;
    @NotNull
    private  LocalDate endDate;
    @NotNull
    private  String bank;
    @NotNull
    private  String instructions;
    private boolean status;
    private String  url;
    @OneToMany(mappedBy="products")
    private Set<Payment> payments;

}
